### ISSUE TITLE
Enable/Disable watering programmes

### ISSUE THREAD

**Issue by CloudWolf99** [sebr/bhyve-home-assistant#14]
Possibility to enable/disable the watering Programmes

**Comment by texiun** [comment]
I'm currently using a template to return the current program with a value template. You could use a switch based on this? 

Considering there is only a possibility of 3 programs A,B,C (AFAIK) Could you base it around this? Direct the switch to only be visable based on the program shown? IE, IF Prog A active do not show B/C? 

`        value_template: >-
                    {%- if states.switch.garden_zone.attributes["program_a"]["enabled"] == true -%}
                    {{ states.switch.garden_zone.attributes["program_a"]["name"] }}
                    {%- elif states.switch.garden_zone.attributes["program_b"]["enabled"] == true -%}
                    {{ states.switch.garden_zone.attributes["program_b"]["name"] }}
                    {%- else -%}
                    None
                    {%- endif -%}`

**Comment by sebr** [comment]
I have a prototype with this working — if you feel up to testing version [0.2.0-alpha.1](https://github.com/sebr/bhyve-home-assistant/releases/tag/0.2.0-alpha.1) that would be much appreciated!

You’ll need to enable beta releases in HACS to install it.

**Comment by texiun** [comment]
> I have a prototype with this working — if you feel up to testing version [0.2.0-alpha.1](https://github.com/sebr/bhyve-home-assistant/releases/tag/0.2.0-alpha.1) that would be much appreciated!
> 
> You’ll need to enable beta releases in HACS to install it.

Just installed it, Should it be listed as a new switch or should i need to reference the program name?

**Comment by sebr** [comment]
They should be new switches. Make sure you restart HA, new HACS is a bit confusing like that

**Comment by texiun** [comment]
> They should be new switches. Make sure you restart HA, new HACS is a bit confusing like that

Looks like it's creating everything on setup nicely! 
```
2020-05-25 23:14:44 INFO (MainThread) [custom_components.bhyve.switch] Creating switch: Garden Zone                                                                
2020-05-25 23:14:44 INFO (MainThread) [custom_components.bhyve.switch] Garden Zone watering_status: None                                                           
2020-05-25 23:14:44 INFO (MainThread) [custom_components.bhyve.switch] Watering program Home (a) is not enabled, skipping                                          
2020-05-25 23:14:44 INFO (MainThread) [custom_components.bhyve.switch] Watering program Away (b) is not enabled, skipping                                          
2020-05-25 23:14:44 INFO (MainThread) [custom_components.bhyve.switch] Creating switch: Program Home                                                               
2020-05-25 23:14:44 INFO (MainThread) [custom_components.bhyve.switch] Creating switch: Program Away                                                               
2020-05-25 23:14:44 INFO (MainThread) [custom_components.bhyve.binary_sensor] Creating binary_sensor: Rain Delay Smart Hose Timer                                  
2020-05-25 23:14:44 INFO (MainThread) [custom_components.bhyve.sensor] Creating sensor: Battery level Smart Hose Timer                                             
2020-05-25 23:14:44 INFO (MainThread) [custom_components.bhyve.sensor] Creating sensor: Garden Zone State                                                          
2020-05-25 23:14:44 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new switch.bhyve entity: switch.home_program                              
2020-05-25 23:14:44 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new switch.bhyve entity: switch.away_program                              
2020-05-25 23:14:44 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new binary_sensor.bhyve entity: binary_sensor.rain_delay_smart_hose_timer_
2                                                                                                                                                                  
2020-05-25 23:14:44 INFO (MainThread) [homeassistant.helpers.entity_registry] Registered new sensor.bhyve entity: sensor.battery_level_smart_hose_timer_2          
2020-05-25 23:14:44 ERROR (MainThread) [homeassistant.components.sensor] Entity id already exists - ignoring: sensor.battery_level_smart_hose_timer_2. Platform bhy
ve does not generate unique IDs```

I'll report back when I have some more info.

**Comment by sebr** [comment]
> `2020-05-25 23:14:44 ERROR (MainThread) [homeassistant.components.sensor] Entity id already exists - ignoring: sensor.battery_level_smart_hose_timer_2. Platform bhyve does not generate unique IDs.`

I'll take a look at this too

**Comment by texiun** [comment]
> > `2020-05-25 23:14:44 ERROR (MainThread) [homeassistant.components.sensor] Entity id already exists - ignoring: sensor.battery_level_smart_hose_timer_2. Platform bhyve does not generate unique IDs.`
> 
> I'll take a look at this too

So everything looks to work pretty well. I've checked all my switches and it updates the info accordingly, pretty nice job that it pulls down the program name! I did notice however, Any changes i'd make Orbit side would take a restart to reflect on HA, Is this by design or is there a polling method timeout you're using?

PS - Is there anywhere I can buy you a coffee?

**Comment by sebr** [comment]
>  I did notice however, Any changes i'd make Orbit side would take a restart to reflect on HA, Is this by design or is there a polling method timeout you're using?

Nope, just haven't finished it yet :)

> PS - Is there anywhere I can buy you a coffee?

That's so kind of you, thanks! Here you can [buy me a coffee](https://www.buymeacoffee.com/sebr)!

**Comment by sebr** [comment]
This has been implemented and released in version [0.2.0-alpha.1](https://github.com/sebr/bhyve-home-assistant/releases/tag/0.2.0-alpha.1)
